File: app/src/main/java/com/kieran/winnipegbusbackend/UpcomingStops/UpcomingStopsManager.java
package com.kieran.winnipegbusbackend.UpcomingStops;

import com.kieran.winnipegbusbackend.LoadResult;
import com.kieran.winnipegbusbackend.RouteKey;

import java.util.ArrayList;

public interface UpcomingStopsManager {
    void GetUpcomingStopsAsync(RouteKey key, int stopOnRoute, OnUpcomingStopsFoundListener listener);

    interface OnUpcomingStopsFoundListener {
        void OnUpcomingStopsFound(LoadResult<ArrayList<Integer>> stops);
    }
}

File: app/src/main/java/com/kieran/winnipegbusbackend/UpcomingStops/UpcomingStopsManager.kt
package com.kieran.winnipegbusbackend.UpcomingStops

import com.kieran.winnipegbusbackend.LoadResult
import com.kieran.winnipegbusbackend.RouteKey

import java.util.ArrayList

interface UpcomingStopsManager {
    fun GetUpcomingStopsAsync(key: RouteKey, stopOnRoute: Int, listener: OnUpcomingStopsFoundListener)

    interface OnUpcomingStopsFoundListener {
        fun OnUpcomingStopsFound(stops: LoadResult<ArrayList<Int>>)
    }
}
